### DISCUSSION TITLE
Custom action in RepeatableEntry

### DISCUSSION THREAD

**Discussion by dzidlicious** [filamentphp/filament#12471] [Help]
### Package

Infolist builder

### Package Version

v 3.0

### How can we help you?

I wonder if there is a way to add action in RepeatableEntry component.  It iterates through records, so I wish to add custom action which deletes this record and another action which changes it's atributes value (bool).

This confuses me because I don't know how to refer to the iterated item. Any help? :-)

I'm attaching code sample below:

return $infolist
            ->state(['replies' => $replies])
            ->schema([
                RepeatableEntry::make('replies')
                    ->label('Showing all messages from the thread no.' . ' ' . $this->record->thread_id)
                    ->schema([
                        TextEntry::make('id')->label('Message id'),
                        TextEntry::make('user_id'),
                        TextEntry::make('created_at')->date('Y-m-d h:i'),
                        TextEntry::make('read')

                            ->suffixAction(
                                Action::make('Mark as unread')
                                    ->icon('heroicon-m-clipboard')
                                    ->requiresConfirmation()
                                    ->action(function (array $data,  SupportTicket $item) {
                                        $this->item->read = 0; // ?? how to refer to this atribute ??
                                            $item->save();
                                    })
                            ),

                        Actions::make([
                            Action::make('Delete')
                                ->icon('heroicon-m-trash')
                                ->iconButton()
                                ->action(function(){
                                    dd($this); // ?? how to refer to this object ?? 
                                })

**Comment by goaround** [comment]
Hi @dzidlicious,

I run into the same question. Have you found a solution?

**Comment by scarm01** [comment]
This is working for me:
```php
                        TextEntry::make('created_at')
                            ->label("Date")
                            ->icon('heroicon-m-calendar-days')
                            ->inlineLabel()
                            ->suffixAction(function (Message $message) {
                                return Infolists\Components\Actions\Action::make('delete')
                                    ->name('delete')
                                    ->color('danger')
                                    ->icon('heroicon-o-trash')
                                    ->label("Delete")
                                    ->action(function () use ($message): void {
                                        $message->delete();

                                        Notification::make()
                                            ->title("Message deleted!")
                                            ->success()
                                            ->actions([
                                                \Filament\Notifications\Actions\Action::make('Undo')
                                                    ->action(function (Message $message) {
                                                        $message->restore();
                                                    })
                                            ])
                                            ->send();
                                    });
                            }),
```

**Reply by goaround** [reply]
Is it inside a RepeatableEntry and Message is the relationship?

**Reply by scarm01** [reply]
In the RepeatableEntry

**Comment by goaround** [comment]
I tried it again, and inside a suffixAction, it works indeed. But I still can't create an EditAction or DeleteAction inside a RepeatableEntry to edit or delete the record.

**Comment by deviarte** [comment]
I also tried many things but have not been able to get an EditAction or DeleteAction to work, they all "inherit" the parent resource.
